test/reputation-system/reputation-mining-client/client-auto-functionality.js

```diff
@@ -0,0 +1,445 @@
+/* eslint-disable no-underscore-dangle */
+
+import path from "path";
+import chai from "chai";
+import bnChai from "bn-chai";
+
+import { TruffleLoader } from "@colony/colony-js-contract-loader-fs";
+
+import { DEFAULT_STAKE, MINING_CYCLE_DURATION } from "../../../helpers/constants";
+import {
+  getActiveRepCycle,
+  forwardTime,
+  advanceMiningCycleNoContest,
+  checkSuccessEthers,
+  checkErrorRevertEthers,
+  mineBlock,
+  finishReputationMiningCycle,
+  currentBlock
+} from "../../../helpers/test-helper";
+import {
+  setupColonyNetwork,
+  setupMetaColonyWithLockedCLNYToken,
+  giveUserCLNYTokensAndStake,
+  setupFinalizedTask,
+  fundColonyWithTokens
+} from "../../../helpers/test-data-generator";
+import ReputationMinerClient from "../../../packages/reputation-miner/ReputationMinerClient";
+import ReputationMinerTestWrapper from "../../../packages/reputation-miner/test/ReputationMinerTestWrapper";
+import MaliciousReputationMinerExtraRep from "../../../packages/reputation-miner/test/MaliciousReputationMinerExtraRep";
+
+const { expect } = chai;
+chai.use(bnChai(web3.utils.BN));
+
+const loader = new TruffleLoader({
+  contractDir: path.resolve(__dirname, "..", "..", "..", "build", "contracts")
+});
+
+const realProviderPort = process.env.SOLIDITY_COVERAGE ? 8555 : 8545;
+
+process.env.SOLIDITY_COVERAGE
+  ? contract.skip
+  : contract("Reputation miner client auto enabled functionality", accounts => {
+      const MINER1 = accounts[5];
+      const MINER2 = accounts[6];
+      const MINER3 = accounts[7];
+
+      let colonyNetwork;
+      let repCycle;
+      let reputationMinerClient;
+      let goodClient;
+      let startingBlockNumber;
+
+      const setupNewNetworkInstance = async (_MINER1, _MINER2, _MINER3) => {
+        colonyNetwork = await setupColonyNetwork();
+
+        const { metaColony, clnyToken } = await setupMetaColonyWithLockedCLNYToken(colonyNetwork);
+        await fundColonyWithTokens(metaColony, clnyToken, DEFAULT_STAKE);
+
+        await giveUserCLNYTokensAndStake(colonyNetwork, _MINER1, DEFAULT_STAKE);
+        await giveUserCLNYTokensAndStake(colonyNetwork, _MINER2, DEFAULT_STAKE);
+        await giveUserCLNYTokensAndStake(colonyNetwork, _MINER3, DEFAULT_STAKE);
+        await colonyNetwork.initialiseReputationMining();
+        await colonyNetwork.startNextCycle();
+
+        await advanceMiningCycleNoContest({ colonyNetwork, test: this });
+        await setupFinalizedTask({ colonyNetwork, colony: metaColony, token: clnyToken });
+        await advanceMiningCycleNoContest({ colonyNetwork, test: this });
+        const startingBlock = await currentBlock();
+        startingBlockNumber = startingBlock.number;
+
+        goodClient = new ReputationMinerTestWrapper({
+          loader,
+          realProviderPort,
+          minerAddress: MINER2,
+          useJsTree: true
+          // dbPath: "./reputationStates_good.sqlite"
+        });
+        reputationMinerClient = new ReputationMinerClient({
```

Comment:
What is the meaning of having `reputationMinerClient` opposed to `goodClient`? Seems like the naming could be improved.